[FLINK-24859] document new connectors formats

echauchot opened the pull request:
R: @AHeise as discussed here are the new formats that were not documented:
- new parquet
- new textfile

They are for DataStream API. 

For orc and hive they are targeted only to the table API and were already documented. And parquet for table API was already documented as well.

I also took the liberty to make the format class names uniform (1st commit)

echauchot commented:
@flinkbot run azure

echauchot commented:
@flinkbot run azure

echauchot commented:
@flinkbot run azure

echauchot commented:
@flinkbot run azure

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
```diff
@@ -0,0 +1,99 @@
+-->
+
+
+# Parquet format
```
```suggestion
# Parquet
```

echauchot replied:
the already merged PR on "old" connector formats contain format in all the titles, I think it's better to be coherent

MartijnVisser replied:
Let's fix that in a follow-up ticket then yes :)

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
```diff
@@ -0,0 +1,99 @@
+
+# Parquet format
+
+Flink supports reading [parquet](https://parquet.apache.org/) files and producing [Flink rows](https://nightlies.apache.org/flink/flink-docs-master/api/java/org/apache/flink/types/Row.html).
```
I'm wondering if it makes sense to include 'producing Flink rows' here. Doesn't Flink "just" support reading and writing Parquet files? Rows are just an intermediary concept of Flink itself?

echauchot replied:
I added this precision in comparison with formats that generate `Pojos` or `GenericRecords` etc ...

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
```diff
@@ -0,0 +1,99 @@
+# Parquet format
+
+Flink supports reading [parquet](https://parquet.apache.org/) files and producing [Flink rows](https://nightlies.apache.org/flink/flink-docs-master/api/java/org/apache/flink/types/Row.html).
+To use that format you need to add the dependency below to your project:
```
```suggestion
To use the format you need to add the Flink Parquet dependency to your project.
```

echauchot replied:
agree, done

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
````diff
@@ -0,0 +1,99 @@
+{{< artifact flink-parquet >}}
+```  
+
+The DataStream API is now compatible with both batch and streaming execution modes. 
````
This could imply that adding the dependency is necessary to make the DataStream API compatible with both batch & streaming mode :) I think we can remove this sentence.

echauchot replied:
I think it is important compared to the old formats that were for DataSet only. But I agree, this sentence would be better in a common page and not in the parquet and text format pages.

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
````diff
@@ -0,0 +1,99 @@
+```  
+
+The DataStream API is now compatible with both batch and streaming execution modes. 
+And this format is compatible with the new Source that can be used in both modes.
+Thus, you can use this format in two ways:
+- Bounded read for batch mode
+- Continuous read for streaming mode: monitors a directory for new files that appear 
````
```suggestion
After adding the dependency, you can read and write Parquet data in either the [Batch Execution Mode or Streaming Execution Mode]({{< ref "/docs/dev/datastream/execution_mode" >}}).
```

echauchot replied:
I think it is important for the user to know that for file reading in streaming mode a monitoring of the directory will be in place and the referred doc does not mention this.

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
```diff
@@ -0,0 +1,99 @@
+
+**Bounded read example**:
+
+In this example we create a DataStream containing parquet records as flink Rows. We project the schema to read only certain fields ("f7", "f4" and "f99").  
```
```suggestion
In this example we create a DataStream-based application containing Parquet records as Flink Rows. We project the schema to read only certain fields ("f7", "f4" and "f99").  
```

echauchot replied:
actually the code snippet only creates a DataStream that reads parquet files and all necessary objects (types, format, source) but it does not create the env, the parameters, the sink etc... I think we cannot talk about an application in that case. But fixing the capitals.

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
```diff
@@ -0,0 +1,99 @@
+**Bounded read example**:
+
+In this example we create a DataStream containing parquet records as flink Rows. We project the schema to read only certain fields ("f7", "f4" and "f99").  
+We read records by batches of 500 records. The first boolean parameter specifies if timestamp columns need to be interpreted as UTC 
```
```suggestion
We read records in batches of 500 records. The first boolean parameter specifies if timestamp columns need to be interpreted as UTC.
```

echauchot replied:
done, thanks

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
```diff
@@ -0,0 +1,99 @@
+
+In this example we create a DataStream containing parquet records as flink Rows. We project the schema to read only certain fields ("f7", "f4" and "f99").  
+We read records by batches of 500 records. The first boolean parameter specifies if timestamp columns need to be interpreted as UTC 
+and the second boolean tells the format if the projected parquet fields names are to be interpreted in a case sensitive way.
```
```suggestion
The second boolean instructs the application if the projected Parquet fields names are to be interpreted in a case sensitive way.
```

echauchot replied:
done, thanks

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/parquet.md:
```diff
@@ -0,0 +1,99 @@
+
+**Continuous read example**:
+
+In this example we create a DataStream containing parquet records as flink Rows that will infinitely grow
+as new files are added to the directory. We monitor for new files each second.
+We project the schema to read only certain fields ("f7", "f4" and "f99").  
+We read records by batches of 500 records. The first boolean parameter specifies if timestamp columns need to be interpreted as UTC
+and the second boolean tells the format if the projected parquet fields names are to be interpreted in a case sensitive way.
+There is no need for a watermark strategy as records do not contain event timestamps.
```
```suggestion
In this example we create a DataStream-based application containing Parquet records as Flink Rows that will infinitely grow as new files are added to the directory. We monitor for new files each second.
We project the schema to read only certain fields ("f7", "f4" and "f99").  
We read records in batches of 500 records. The first boolean parameter specifies if timestamp columns need to be interpreted as UTC, 
The second boolean instructs the application if the projected Parquet fields names are to be interpreted in a case sensitive way.
There is no need for a watermark strategy as records do not contain event timestamps.
```

echauchot replied:
done, thanks

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/text_files.md:
```diff
@@ -0,0 +1,69 @@
+-->
+
+
+# Text files format
```
I don't have a better suggestion, but I think `Text files format` is ambiguous. It could be CSV, JSON, AVRO etc. WDYT?

echauchot replied:
I think text is better as this format provides `Strings` that can be parsed as CSV, JSON etc... in a map downstream the pipeline.

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/text_files.md:
````diff
@@ -0,0 +1,69 @@
+# Text files format
+
+Flink supports reading from text lines from a file using `TextLineInputFormat`. This format uses Java's built-in InputStreamReader to decode the byte stream using various supported charset encodings.
+To use that format you need to add the dependency below to your project:
+
+```xml
+{{< artifact flink-connector-files >}}
+```
+
+The DataStream API is now compatible with both batch and streaming execution modes.
+And this format is compatible with the new Source that can be used in both modes.
+Thus, you can use this format in two ways:
+- Bounded read for batch mode
+- Continuous read for streaming mode: monitors a directory for new files that appear
+
+**Bounded read example**:
+
+In this example we create a DataStream containing the lines of a text file as Strings. 
+There is no need for a watermark strategy as records do not contain event timestamps.
````
My feedback from earlier also applies to this paragraph.

echauchot replied:
ditto

MartijnVisser commented on docs/content/docs/connectors/datastream/formats/text_files.md:
````diff
@@ -0,0 +1,69 @@
+  env.fromSource(source, WatermarkStrategy.noWatermarks(), "file-source");
+```
+
+**Continuous read example**:
+In this example, we create a DataStream containing the lines of text files as Strings that will infinitely grow 
+as new files are added to the directory. We monitor for new files each second.
+There is no need for a watermark strategy as records do not contain event timestamps.
````
My feedback from earlier also applies to this paragraph.

echauchot replied:
ok

MartijnVisser reviewed (changes requested):
Thanks a lot @echauchot for this PR! It's a welcome addition. I've left some review comments on the documentation; I don't think I'm best suited to do the Java based reviews in this case.

echauchot commented:
@flinkbot run azure

echauchot commented:
@flinkbot run azure

echauchot commented:
@flinkbot run azure

echauchot commented:
@MartijnVisser rebased on master. And applying your comments....

echauchot commented:
@AHeise once this PR is merged, I'll backport it to 1.14 as for the other PR.

echauchot commented:
@MartijnVisser @AHeise do you agree with my commit that renames formats to make the name coherent ? It would be a breaking change as they were already released. But yet, they were not documented so it might not be such a big deal. WDYT ?

echauchot commented:
@MartijnVisser thanks for your review once again! I addressed your comments and resolved the related discussions. For your questions I answered and left discussions open for you to see the answers. But there are some of your suggestions for which I don't totally agree, I left these discussions open. PTAL.

MartijnVisser reviewed (approved):
Thanks a lot! I don't see any objections to merge this. Any improvements we can always pick up later when users start reading/provide feedback on the documentation.

MartijnVisser commented:
@AHeise Please merge

echauchot commented:
@MartijnVisser thanks for your review !

AHeise commented:
Thank you again for your contribution!

echauchot commented:
Thanks for merging @AHeise !
